src/example_package/gui/app.py

```diff
@@ -36,8 +39,28 @@ class MainScreen(BoxLayout):
 
     def __init__(self, **kwargs):
         super(MainScreen, self).__init__(**kwargs)
+        self.chosen_participants = pd.DataFrame()
         self.update_content('Quick Start')
 
+    def update_contestants(self, df):
+        self.chosen_participants = df
+
+    def clear_contestants(self):
+        self.chosen_participants = pd.DataFrame()
+
+    def is_selected(self, participant):
+        if any(self.chosen_participants.iloc[i, :].equals(participant) for i in range(len(self.chosen_participants))):
+            return True
+        return False
+
+    def toggle_participant(self, participant):
```

Comment:
Dotychczas wskazywałem jakieś mało znaczące dorbiazgi, z którymi szybko sobie poradzisz, ale to będzie największa rzecz.

Obecnie klasa `MainScreen` zajmuje się w całości zarządzaniem listą wybranych zawodników, co chyba w sumie nie powinno być zadaniem klasy do wyświetlania GUI aplikacji XD. Łamana jest 'Single Responsibility Principle'.

Może dałoby radę przenieść metody `update_contestants`, `clear_contestants`, `is_selected` i `toggle_participant` razem z polem `self.chosen_participants` do jakiejś osobnej, bardziej backendowej klasy? Typu np. `ChosenParticipantsManager` czy coś w tym stylu.
Wtedy w `MainScreen` możemy trzymać takiego managera w polu 
```
self.chosen_participants_manager = ChosenParticipantsManager(...)
```
 i reszta kodu nie powinna ulec zmianie. A zyskamy wiele, bo rozdzielimy odpowiedzialności między klasy.

Co myślicie? Zapraszam do dyskusji :)

Reply:
Mocne +1